mesonbuild/environment.py

```diff
@@ -1268,14 +1268,8 @@ def from_config_parser(cls, parser: configparser.ConfigParser):
         return out
 
 class Properties:
-    def __init__(self):
-        self.properties = {}
-
-    def get_external_args(self, language):
-        return mesonlib.stringlistify(self.properties.get(language + '_args', []))
-
-    def get_external_link_args(self, language):
-        return mesonlib.stringlistify(self.properties.get(language + '_link_args', []))
+    def __init__(self, properties: dict = {}):
```

Comment:
I've got a couple of problems with this, for one this should be using `typing.Dict`, with key types. preferably we know what the types are, or at least what types the keys are. second, don't use mutable types as keyword arguments, it's bad. it's really bad.

Here's the clever way to handle this:
```python
def __init__(self, properties: typing.Optional[typing.Dict[str, typing.Any]] = None):
    ...
    self.properties = properties or {}
```